### PR TITLE
UTAG-4580

### PR DESCRIPTION
**Pull request by CamiloAFlorez**

### CODE REVIEW

File: utag-console-persistence/src/main/java/com/cadreon/utag/console/entity/utag/converters/ActionConverter.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.cadreon.utag.console.entity.utag.converters;
```
Comment (mcabras) [inline utag-console-persistence/src/main/java/com/cadreon/utag/console/entity/utag/converters/ActionConverter.java]: Check to see if this class is not needed

File: utag-console-persistence/src/main/java/com/cadreon/utag/console/entity/utag/converters/ModelTypeConverter.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package com.cadreon.utag.console.entity.utag.converters;
```
Comment (mcabras) [inline utag-console-persistence/src/main/java/com/cadreon/utag/console/entity/utag/converters/ModelTypeConverter.java]: Check to see if this class is not needed

File: utag-console-persistence/pom.xml [hunk 3]
```diff
@@ -19,6 +19,11 @@
 	</properties>
 
 	<dependencies>
+		<dependency>
+			<groupId>com.cadreon.utag.console</groupId>
+			<artifactId>utag-console-common</artifactId>
+			<version>${project.version}</version>
+		</dependency>
```
Comment (mcabras) [inline utag-console-persistence/pom.xml]: Remove dependency

File: utag-console-apis/src/main/java/com/cadreon/utag/console/controllers/VariableController.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+@RestController
+@RequestMapping("/tag/{idTag}/variable")
+public class VariableController extends AbstractController {
+
+  @Autowired
+  private VariableService variableService;
+
+  @RequestMapping(method = RequestMethod.GET)
+  public List<VariableDto> searchVariablesforTag(@PathVariable(value = "idTag", required = true) Long totaltagId,
+      @RequestParam(value = "searchString", required = false) String searchString, @RequestParam(value = "all", required = false) Boolean all) {
```
Comment (mcabras) [inline utag-console-apis/src/main/java/com/cadreon/utag/console/controllers/VariableController.java]: remove unused variable "all"

File: utag-console-apis/src/main/java/com/cadreon/utag/console/controllers/VariableController.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+  @RequestMapping(method = RequestMethod.POST)
+  public VariableDto createVariable(@PathVariable(value = "idTag", required = true) Long totaltagId, @Valid @RequestBody VariableDto dto) throws InvalidRequestException {
+    return variableService.createVariable(totaltagId, dto);
+
+  }
+
+  @RequestMapping(value = "/{variableId}", method = RequestMethod.DELETE)
+  public ResponseEntity<Void> deleteVariable(@PathVariable(value = "variableId", required = true) Long variableId, HttpServletRequest request) throws InvalidRequestException {
+    variableService.deleteVariable(variableId);
```
Comment (mcabras) [inline utag-console-apis/src/main/java/com/cadreon/utag/console/controllers/VariableController.java]: When returning void, FE will no be able to determine id it was done correctly or not, please return a response accordingly

File: utag-console-services/src/main/java/com/cadreon/utag/console/services/impl/VariableServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+      variableRepository.delete(variable);
+      History history = History.builder()
+          .action(Action.destroy)
+          .modelId(variable.getId())
+          .name(variable.getKey())
+          .username(user.getLogin())
+          .modelType(ModelType.Variable)
+          .isBulk(false)
+          .build();
+
```
Comment (mcabras) [inline utag-console-services/src/main/java/com/cadreon/utag/console/services/impl/VariableServiceImpl.java]: totalTagId is not getting persisted

File: utag-console-services/src/main/java/com/cadreon/utag/console/services/impl/VariableServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+      dto.setMappedPixelCount(pixelService.findCountByTotalTagIdAndKey(dto.getTotalTagId(), dto.getKey()));
+    }
+
+    return list;
+  }
+
+  @Override
+  public List<StandardVariableDto> getStandardMacros() {
+    List<StandardVariable> standardVariables = standardVariableRepository.findAll();
+    return mapper.mapAsList(standardVariables, StandardVariableDto.class);
```
Comment (mcabras) [inline utag-console-services/src/main/java/com/cadreon/utag/console/services/impl/VariableServiceImpl.java]: What happens if standardVariables is null?

File: utag-console-apis/src/main/java/com/cadreon/utag/console/controllers/VariableController.java (hunk 4 above)
Comment (mcabras) [inline utag-console-apis/src/main/java/com/cadreon/utag/console/controllers/VariableController.java]: Pagination is missing

Comment (mcabras) [inline utag-console-apis/src/main/java/com/cadreon/utag/console/controllers/VariableController.java]: When list is empty it should at least say if the process had finished successfully or not

File: utag-console-services/src/main/java/com/cadreon/utag/console/services/impl/VariableServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+  @Transactional(propagation = Propagation.REQUIRED)
+  public void deleteVariable(Long variableId) throws InvalidRequestException {
+    Variable variable = variableRepository.findOne(variableId);
+    if (variable == null) {
+      throw new InvalidRequestException("Variable with Id: " + variableId + " doesn't exist");
+    }
+
+    Long mappedPixelCount = pixelService.findCountByTotalTagIdAndKey(variable.getTotalTagId(), variable.getKey());
+    Wso2User user = userService.getCurrentUser().orElseThrow(() -> new InvalidRequestException("Security error"));
+
```
Comment (mcabras) [inline utag-console-services/src/main/java/com/cadreon/utag/console/services/impl/VariableServiceImpl.java]: unnecessary use of Wso2User

File: utag-console-services/src/main/java/com/cadreon/utag/console/services/impl/VariableServiceImpl.java (hunk 7 above)
Comment (CamiloAFlorez) [reply utag-console-services/src/main/java/com/cadreon/utag/console/services/impl/VariableServiceImpl.java]: Return an empty List

**Pull request merged**